### PR TITLE
SWATCH-1917: Set cores/sockets to null when value is not present

### PR DESCRIPTION
**Pull request by kflahert**
Jira issue: SWATCH-1917

## Description

In order to have consistency between instance_measurements and host_tally_buckets we need to set the cores/sockets to null when no value it present. Previously these were defaulted to 0.

### Setup

1.Insert test data: 
```
INSERT INTO public.events VALUES ('942e5a9b-35b3-4caf-9d31-5544b1591377', '2023-10-31 03:00:00+00', '{"sla": "Premium", "role": "moa-hostedcontrolplane", "org_id": "org123", "event_id": "942e5a9b-35b3-4caf-9d31-5544b1591377", "timestamp": "2023-10-31T03:00:00Z", "event_type": "snapshot_rosa_cores", "expiration": "2023-10-31T04:00:00Z", "instance_id": "10c6e649-8985-4809-ace3-97a55a615fc3", "display_name": "10c6e649-8985-4809-ace3-97a55a615fc3", "event_source": "prometheus", "measurements": [{"uom": "Cores", "value": 4.0}], "service_type": "rosa Instance", "billing_provider": "aws", "metering_batch_id": "45136d97-6963-4adb-8c25-c2352e0ffe63", "billing_account_id": "463992265265"}', 'snapshot_rosa_cores', 'prometheus', '10c6e649-8985-4809-ace3-97a55a615fc3', 'org123', DEFAULT, '45136d97-6963-4adb-8c25-c2352e0ffe63', '2023-10-31 04:30:25.486382+00');
INSERT INTO public.events VALUES ('db5448c2-19cc-41d3-86e4-956dfc25d8fb', '2023-10-04 03:00:00+00', '{"sla": "Premium", "role": "moa-hostedcontrolplane", "org_id": "org123", "event_id": "db5448c2-19cc-41d3-86e4-956dfc25d8fb", "timestamp": "2023-10-04T03:00:00Z", "event_type": "snapshot_rosa_cores", "expiration": "2023-10-04T04:00:00Z", "instance_id": "39146518-51c2-41a1-881e-1fa007facbd0", "display_name": "39146518-51c2-41a1-881e-1fa007facbd0", "event_source": "prometheus", "measurements": [{"uom": "Sockets", "value": 10}], "service_type": "rosa Instance", "account_number": "6344158", "billing_provider": "aws", "metering_batch_id": "da2c3b9c-2ab8-4b6b-934b-536fc8614625", "billing_account_id": "463992265265"}', 'snapshot_rosa_cores', 'prometheus', '39146518-51c2-41a1-881e-1fa007facbd0', 'org123', DEFAULT, 'da2c3b9c-2ab8-4b6b-934b-536fc8614625', '2023-10-04 04:32:09.229067+00');
```

### Steps

1. Start app with: `DEV_MODE=true CONTRACT_USE_STUB=true SPRING_PROFILES_ACTIVE=worker,api,kafka-queue ./gradlew :bootRun`
1. Run tally:` http POST ":8000/api/rhsm-subscriptions/v1/internal/tally/hourly?org=org123&start=2023-10-01T08:00:00Z&end=2023-11-02T08:00:00Z" x-rh-swatch-psk:placeholder`

### Verification

1. Check host_tally_buckets table for a set of records with null sockets and another with null cores:
`select * from host_tally_buckets;`

### CODE REVIEW

**Review by kahowell — APPROVED** [review]

File: src/main/java/org/candlepin/subscriptions/tally/collector/RHELProductUsageCollector.java [hunk 1]
```diff
@@ -54,8 +54,8 @@ public void collectForHypervisor(UsageCalculation prodCalc, NormalizedFacts hype
   @Override
   public Optional<HostTallyBucket> buildBucket(
       UsageCalculation.Key key, NormalizedFacts normalizedFacts) {
-    int appliedCores = Optional.ofNullable(normalizedFacts.getCores()).orElse(0);
-    int appliedSockets = Optional.ofNullable(normalizedFacts.getSockets()).orElse(0);
+    Integer appliedCores = normalizedFacts.getCores();
+    Integer appliedSockets = normalizedFacts.getSockets();
```
Comment (kahowell) [inline src/main/java/org/candlepin/subscriptions/tally/collector/RHELProductUsageCollector.java:58]: I think we need to make a similar change to the DefaultProductUsageCollector.

**Review by kahowell — CHANGES_REQUESTED** [review]
See inline comment.

**Review by kahowell — APPROVED** [review]

**Review by ssidhaye — APPROVED** [review]
After generating events via test code, and tallying I am able to see null / no value for sockets / core where they are not applicable.

```
host_id                               |   usage    | product_id |   sla   | as_hypervisor | cores | sockets | measurement_type | billing_provider | billing_account_id | version 
 3debcefc-cb0c-417c-9ba1-23119a4ca08a | Production | rosa       | Premium | f             |     4 |         |                  | _ANY             | 463992265265       |       0
 3debcefc-cb0c-417c-9ba1-23119a4ca08a | Production | rosa       | Premium | f             |     4 |         |                  | aws              | _ANY               |       0
 3debcefc-cb0c-417c-9ba1-23119a4ca08a | Production | rosa       | Premium | f             |     4 |         |                  | aws              | 463992265265       |       0
 6b0e37e6-2b08-4b84-848e-0837fe80cfa7 | _ANY       | rosa       | Premium | f             |       |      10 |                  | _ANY             | _ANY               |       0
 6b0e37e6-2b08-4b84-848e-0837fe80cfa7 | _ANY       | rosa       | Premium | f             |       |      10 |                  | _ANY             | 463992265265       |       0
 6b0e37e6-2b08-4b84-848e-0837fe80cfa7 | _ANY       | rosa       | Premium | f             |       |      10 |                  | aws              | _ANY               |       0
```

**Comment by ssidhaye** [comment]
@kflahert I have added a fix for the failing test. But we are hitting an issue due to a [bug](https://issues.redhat.com/browse/SWATCH-2138)

**Comment by kahowell** [comment]
@ssidhaye can you please update tests as necessary?

We're still getting failures in:

-  `tests.tally.test_tally_openshift.test_validate_tally_on_openshift_cores[2]`
-  `tests.tally.test_tally_with_sla_filters.test_validate_sla_filters_for_openshift_tally[Disaster Recovery-Premium-Cores/vCPU]`

**Pull request merged**